drivers/gpio/gpio_nct38xx_alert.c

```diff
@@ -0,0 +1,128 @@
+	/* Alert worker */
+	struct k_work alert_worker;
+};
+
+/* Driver convenience defines */
+#define DRV_CONFIG(dev) ((const struct nct38xx_alert_config *)(dev)->config)
+#define DRV_DATA(dev) ((struct nct38xx_alert_data *)(dev)->data)
+
+static void nct38xx_alert_callback(const struct device *dev, struct gpio_callback *cb,
+				   uint32_t pins)
+{
+	ARG_UNUSED(pins);
+	struct nct38xx_alert_data *data = CONTAINER_OF(cb, struct nct38xx_alert_data, gpio_cb);
+
+	k_work_submit(&data->alert_worker);
+}
+
+static void nct38xx_alert_worker(struct k_work *work)
+{
+	struct nct38xx_alert_data *const data =
+		CONTAINER_OF(work, struct nct38xx_alert_data, alert_worker);
+	const struct nct38xx_alert_config *const config = DRV_CONFIG(data->alert_dev);
+
+	do {
+		/* nct38xx device handler */
+		for (int i = 0; i < config->nct38xx_num; i++) {
+			nct38xx_gpio_alert_handler(config->nct38xx_dev[i]);
+		}
+		/* While the interrupt signal is still active; we have more work to do. */
+	} while (gpio_pin_get(config->irq_gpio_dev, config->irq_gpio_pin));
+}
+
+static int nct38xx_alert_init(const struct device *dev)
+{
+	const struct nct38xx_alert_config *const config = DRV_CONFIG(dev);
+	struct nct38xx_alert_data *const data = DRV_DATA(dev);
+	int ret;
+
+	/* Check nct38xx devices are all ready. */
+	for (int i = 0; i < config->nct38xx_num; i++) {
+		if (!device_is_ready(config->nct38xx_dev[i])) {
+			LOG_ERR("%s device not ready", config->nct38xx_dev[i]->name);
+			return -ENODEV;
+		}
+	}
+
+	/* Set the alert pin for handling the interrupt */
+	k_work_init(&data->alert_worker, nct38xx_alert_worker);
+
+	if (!device_is_ready(config->irq_gpio_dev)) {
+		LOG_ERR("%s device not ready", config->irq_gpio_dev->name);
+		return -ENODEV;
+	}
+
+	gpio_pin_configure(config->irq_gpio_dev, config->irq_gpio_pin,
+			   GPIO_INPUT | config->irq_gpio_flags);
+
+	gpio_init_callback(&data->gpio_cb, nct38xx_alert_callback, BIT(config->irq_gpio_pin));
+
+	ret = gpio_add_callback(config->irq_gpio_dev, &data->gpio_cb);
+	if (ret < 0) {
+		return ret;
+	}
+
+	gpio_pin_interrupt_configure(config->irq_gpio_dev, config->irq_gpio_pin,
+				     GPIO_INT_EDGE_TO_ACTIVE);
+
+	return 0;
+}
+
+#define NCT38XX_DEV_AND_COMMA(node_id, prop, idx)                                                  \
+	DEVICE_DT_GET(DT_PHANDLE_BY_IDX(node_id, nct38xx_dev, idx)),
+
+#define NCT38XX_ALERT_DEVICE_INSTANCE(inst)                                                        \
+	const struct device *nct38xx_dev_##inst[] = { DT_INST_FOREACH_PROP_ELEM(                   \
+		inst, nct38xx_dev, NCT38XX_DEV_AND_COMMA) };                                       \
+	static const struct nct38xx_alert_config nct38xx_alert_cfg_##inst = {                      \
+		.irq_gpio_dev = DEVICE_DT_GET(DT_GPIO_CTLR(DT_DRV_INST(inst), irq_gpios)),         \
+		.irq_gpio_pin = DT_INST_GPIO_PIN(inst, irq_gpios),                                 \
+		.irq_gpio_flags = DT_INST_GPIO_FLAGS(inst, irq_gpios),                             \
```

Comment:
Please use a gpio_dt_spec instead, and initialize it with GPIO_DT_SPEC_INST_GET.

Reply:
Done.